src/iuse.cpp

```diff
@@ -9187,6 +9187,77 @@ int iuse::disassemble( player *p, item *it, bool, const tripoint & )
     return 0;
 }
 
+int iuse::gobag( player *p, item *it, bool, const tripoint & )
+{
+    std::map<std::string, int> gobag_contents = {
+        { "helmet_army", 1 },
+        { "jacket_army", 1 },
+        { "pants_army", 1 },
+        { "tshirt", 3 },
+        { "under_armor", 1 },
+        { "chestrig", 1 },
+        { "coat_rain", 1 },
+        { "hood_rain", 1 },
+        { "long_underpants", 1 },
+        { "towel", 1 },
+        { "boots_combat", 1 },
+        { "socks", 3 },
+        { "gloves_tactical", 1 },
+        { "light_plus_battery_cell", 1 },
+        { "flashlight", 1 },
+        { "radio", 1 },
+        { "lighter", 1 },
+        { "pockknife", 1 },
+        { "multitool", 1 },
+        { "1st_aid", 1 },
+        { "rollmat", 1 },
+        { "sleeping_bag_roll", 1 },
+        { "ear_plugs", 1 },
+        { "mre_hashbrownbacon_box", 1 },
+        { "mre_macaronimarinara_box", 1 },
+        { "mre_meatball_box", 1 },
+        { "bottle_plastic", 2 }
+    };
+
+    p->add_msg_if_player( _( "You empty the contents of the go bag onto the floor." ) );
+
+    item duffelbag = item( "duffelbag" );
+    item backpack = item( "backpack" );
+    if( it->has_flag( "FILTHY" ) ) {
+        duffelbag.set_flag( "FILTHY" );
+        backpack.set_flag( "FILTHY" );
+    }
+    g->m.add_item_or_charges( p->posx(), p->posy(), duffelbag );
+    g->m.add_item_or_charges( p->posx(), p->posy(), backpack );
```

Comment:
Please use the 3D overload:
```suggestion
    g->m.add_item_or_charges( p->pos(), backpack );
```
Same above and below.